pastepwn/scraping/pastebin/pastebinscraper.py

```diff
@@ -165,12 +166,13 @@ def start(self, paste_queue):
                 self.running = False
                 break
 
-            # check if time since last
-            current_time = int(time.time())
-            diff = current_time - self._last_scrape_time
+            while self.running:
+                current_time = int(time.time())
+                diff = current_time - self._last_scrape_time
 
-            # if the last scraping happened less than 60 seconds ago,
-            # wait until the 60 seconds passed
-            if diff < 60:
-                sleep_time = 60 - diff
-                time.sleep(sleep_time)
+                if diff > 60:
+                    break
+
+                # if the last scraping happened less than 60 seconds ago,
+                # wait 2 seconds and check again
+                time.sleep(2)
```

Comment:
This was done in order to prevent very long wait times when using CTRL-C, since this operation is blocking